### PR TITLE
Fix Json serialization error

### LINKED ISSUE #1883: Fresh install with code from master doesn't run (error creating UserRootFolder)
**Describe the bug**

Jellyfin is unable to start with a fresh install, and no data ever set. An exception is immediately thrown on startup mentioning " Error creating UserRootFolder e9d5075a-555c-1cbc-394e-ec4cef295274". A quick look with strace seems to indicate that something crashes on the w ay.

**To Reproduce**
1. Get the latest master from jellyfin and jellyfin-web
2. Build and install to a path of liking
3. Start jellyfin with a fresh data directory (empty)

**Expected behavior**
Jellyfin starts.

**Logs**

```
[07:30:29] [INF] 64-Bit Process: True                                                                                                                                                                                                               [0/28398]
[07:30:29] [INF] User Interactive: True                                                                                       
[07:30:29] [INF] Processor count: 4                                                                                           
[07:30:29] [INF] Program data path: /home/jellyfin/jellyfin-data/                                                             
[07:30:29] [INF] Web resources path: /home/jellyfin/jellyfin-build/jellyfin-web                                               
[07:30:29] [INF] Application directory: /home/jellyfin/jellyfin-build/                                                        
[07:30:29] [INF] Setting cache path to /home/jellyfin/jellyfin-data/cache                                                     
[07:30:30] [INF] Loading assemblies                                                                                           
[07:30:30] [ERR] Error creating UserRootFolder e9d5075a-555c-1cbc-394e-ec4cef295274                                           
System.NullReferenceException: Object reference not set to an instance of an object.                                                                                                                                                                         
   at Emby.Server.Implementations.Library.LibraryManager.GetItemById(Guid id)                                                 
   at Emby.Server.Implementations.Library.LibraryManager.GetUserRootFolder()                       
[07:30:30] [FTL] Error while starting server.                                                                                                                                                                                                                
System.ArgumentNullException: Value cannot be null.                                                                           
Parameter name: source                                                                                                                                                                                                                                       
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)                                                                                                                                                                            
   at Emby.Server.Implementations.Library.LibraryManager.ResolvePath(FileSystemMetadata fileInfo, IDirectoryService directoryService, IItemResolver[] resolvers, Folder parent, String collectionType, LibraryOptions libraryOptions)                        
   at Emby.Server.Implementations.Library.LibraryManager.GetUserRootFolder()                                
   at System.Text.Json.JsonPropertyInfoCommon`4.GetValueAsObject(Object obj)                                                  
   at System.Text.Json.JsonSerializer.HandleObject(JsonPropertyInfo jsonPropertyInfo, JsonSerializerOptions options, Utf8JsonWriter writer, WriteStack& state)                                                                                               
   at System.Text.Json.JsonSerializer.WriteObject(JsonSerializerOptions options, Utf8JsonWriter writer, WriteStack& state)    
   at System.Text.Json.JsonSerializer.Write(Utf8JsonWriter writer, Int32 originalWriterDepth, Int32 flushThreshold, JsonSerializerOptions options, WriteStack& state)                                                                                        
   at System.Text.Json.JsonSerializer.WriteCore(Utf8JsonWriter writer, PooledByteBufferWriter output, Object value, Type type, JsonSerializerOptions options)                                                                                                
   at System.Text.Json.JsonSerializer.WriteCore(PooledByteBufferWriter output, Object value, Type type, JsonSerializerOptions options)                                                                                                                       
   at System.Text.Json.JsonSerializer.WriteCoreBytes(Object value, Type type, JsonSerializerOptions options)                  
   at Emby.Server.Implementations.Data.SqliteUserRepository.CreateUser(User user)                                             
   at Emby.Server.Implementations.Library.UserManager.LoadUsers()                                                                                                                                                                                            
   at Emby.Server.Implementations.Library.UserManager.Initialize()                                                                                                                                                                                           
   at Emby.Server.Implementations.ApplicationHost.RegisterResources(IServiceCollection serviceCollection)                     
   at Emby.Server.Implementations.ApplicationHost.InitAsync(IServiceCollection serviceCollection)                                                                                                                                                            
   at Jellyfin.Server.Program.StartApp(StartupOptions options)
[07:30:30] [INF] Disposing CoreAppHost
[07:30:30] [INF] Received a SIGTERM signal, shutting down 
```

**System (please complete the following information):**
 - OS: openSUSE Tumbleweed
 - Jellyfin Version: latest master

### PR DESCRIPTION
**Pull request by Bond-009**
Should fix #1883

### CODE REVIEW

**Comment by Bond-009** [comment]
@JustAMan Done

File: Emby.Server.Implementations/Services/ServicePath.cs [hunk 1]
```diff
@@ -190,21 +198,12 @@ public RestPath(Func<Type, object> createInstanceFn, Func<Type, Func<string, obj
                     StringComparer.OrdinalIgnoreCase);
         }
 
-        internal static string[] IgnoreAttributesNamed = new[]
-        {
-            "IgnoreDataMemberAttribute",
```
Comment (JustAMan) [inline Emby.Server.Implementations/Services/ServicePath.cs]: why is this no longer needed? I see you transferred `JsonIgnoreAttribute` to a new position

Comment (Bond-009) [reply Emby.Server.Implementations/Services/ServicePath.cs]: The IgnoreDataMemberAttribute got removed, and I moved the field up above the ctor.

**Comment by joshuaboniface** [comment]
Cherry-picked into 10.4.1.

**Comment by dkanada** [comment]
@Bond-009 was this needed in the 10.4.1 release? I thought you mentioned the original bug was introduced after 10.4.0 so we didn't need to backport this.

**Comment by Bond-009** [comment]
Correct, we don’t really need to, but apparently it doesn’t break anything. The only disadvantage is that it adds another dep.

**Comment by joshuaboniface** [comment]
@Bond-009 @dkanada it does seem to fix the login issue though, which was definitely an issue in 10.4.0 unless I'm really mistaken.

**Comment by gerroon** [comment]
I think I am hit by this bug, does anyone know when the nightly is coming out? I have been out for weeks :(

**Pull request merged**